cli/index.js

```diff
@@ -1117,17 +1117,22 @@ async function getConfig(file, baseDir, readFile) {
 }
 
 /** Checks diagnostics emitted so far for errors. */
-export function checkDiagnostics(program, stderr, reportDiagnostic, useColors) {
+export function checkDiagnostics(program, stderr, disableWarning, reportDiagnostic, useColors) {
   if (typeof useColors === "undefined" && stderr) useColors = stderr.isTTY;
   var numErrors = 0;
   do {
     let diagnostic = assemblyscript.nextDiagnostic(program);
     if (!diagnostic) break;
     if (stderr) {
-      stderr.write(
-        assemblyscript.formatDiagnostic(diagnostic, useColors, true) +
-        EOL + EOL
-      );
+      const checkWarningPrint = (diagnostic) => {
```

Comment:
A slightly better name might be `isDisabledWarning`, inverting the check.

Reply:
I think with the name now changed, the logic of the helper function should be inverted so it returns `true` if a warning **is** disabled, where the check below then becomes `|| !isDisabledWarning(...)`.